src/sonic-yang-models/yang-models/sonic-vrrp.yang

```diff
@@ -0,0 +1,283 @@
+module sonic-vrrp {
+    namespace "http://github.com/Azure/sonic-vrrp";
+    prefix svrrp;
+    yang-version 1.1;
+
+    import ietf-inet-types {
+        prefix "inet";
+    }
+
+    import sonic-interface {
+        prefix sint;
+    }
+
+    import sonic-port {
+        prefix sprt;
+    }
+
+    import sonic-portchannel {
+        prefix spc;
+    }
+
+    import sonic-vlan-sub-interface {
+        prefix svlansubint;
+    }    
+
+    import sonic-vlan {
+        prefix svlan;
+    }
+
+    organization
+        "SONiC";
+
+    contact
+        "SONiC";
+
+    description
+        "SONIC VRRP";
+
+    revision 2024-11-01 {
+        description
+            "Initial revision.";
+    }
+
+    container sonic-vrrp {
+
+        container VRRP {
+            description
+            "Configures VRRP version 2 or 3 for IPv4.";
+
+            list VRRP_LIST {
+                max-elements 128;
+                key "ifname idkey";
+
+                leaf ifname {
+                    type union {
+                        type leafref {
+                            path "/sint:sonic-interface/sint:INTERFACE/sint:INTERFACE_LIST/sint:name";
+                        }
+                        type leafref {
+                            path "/spc:sonic-portchannel-interface/spc:PORTCHANNEL_INTERFACE/spc:PORTCHANNEL_INTERFACE_LIST/spc:name";
+                        }
+                        type leafref {
+                            path "/svlan:sonic-vlan/svlan:VLAN_INTERFACE/svlan:VLAN_INTERFACE_LIST/svlan:name";
+                        }
+                        type leafref {
+                            path "/svlansubint:sonic-vlan-sub-interface/svlansubint:VLAN_SUB_INTERFACE/svlansubint:VLAN_SUB_INTERFACE_LIST/svlansubint:name";
+                        }
+                    }
+                }
+
+                leaf idkey {
+                    type uint8 {
+                    range "1..255";
+                    }
+                }
+
+                leaf vrid {
+                    type uint8 {
+                    range "1..255";
+                    }
+                }
+
+                leaf version {
+                    type string {
+                        pattern '2|3';
+                    }
+                }
+
+                leaf pre_empt {
```

Comment:
Is there a specific reason to name like pre_empt instead of preempt?

Reply:
Just for better readability.